crates/holochain/src/conductor/interface/websocket.rs

```diff
@@ -1206,6 +1207,66 @@ mod test {
         conductor_handle.shutdown().await.unwrap().unwrap();
     }
 
+    #[tokio::test(flavor = "multi_thread")]
+    async fn dump_conductor_state() {
+        holochain_trace::test_run();
+        let uuid = Uuid::new_v4();
+        let dna = fake_dna_zomes(
+            &uuid.to_string(),
+            vec![("zomey".into(), TestWasm::Foo.into())],
+        );
+        let agent_pubkey = fake_agent_pubkey_1();
+
+        let _ = RealRibosomeFixturator::new(crate::fixt::curve::Zomes(vec![TestWasm::Foo]))
+            .next()
+            .unwrap();
+        let (_tmpdir, _, conductor_handle, _agent_key) = setup_app_in_new_conductor(
+            "test app".to_string(),
+            Some(agent_pubkey.clone()),
+            vec![(dna.clone(), None)],
+        )
+        .await;
+        let dna_hash = dna.dna_hash();
+
+        // Allow agents time to join
+        tokio::time::sleep(std::time::Duration::from_secs(2)).await;
+
+        #[derive(Serialize, Debug)]
+        pub struct ConductorSerialized {
+            running_cells: Vec<(DnaHashB64, AgentPubKeyB64)>,
+            shutting_down: bool,
+            admin_websocket_ports: Vec<u16>,
+            app_interfaces: Vec<AppInterfaceId>,
+        }
+
+        #[derive(Serialize, Debug)]
+        struct ConductorDump {
+            conductor: ConductorSerialized,
+            state: ConductorState,
+        }
+
+        let expected = ConductorDump {
+            conductor: ConductorSerialized {
+                running_cells: vec![(dna_hash.clone().into(), agent_pubkey.clone().into())],
+                shutting_down: false,
+                admin_websocket_ports: vec![],
+                app_interfaces: vec![],
+            },
+            state: conductor_handle.get_state_from_handle().await.unwrap(),
```

Comment:
This still relies on calling `conductor.get_state()` to generate the expected response.